docs/source/application_development/porter.rst

```diff
@@ -0,0 +1,633 @@
+| **Parameter**                    | **Type**      | **Description**                        |
++==================================+===============+========================================+
+| ``treasure_map_id``              | String        | Treasure map identifier.               |
++----------------------------------+---------------+----------------------------------------+
+| ``bob_encrypting_key``           | String        | Bob's encrypting key encoded as hex.   |
++----------------------------------+---------------+----------------------------------------+
+
+
+Returns
++++++++
+The requested treasure map:
+
+    * ``treasure_map`` - Treasure map bytes encoded as base64
+
+Example Request
++++++++++++++++
+.. code:: bash
+
+    curl -X GET <PORTER URI>/get_treasure_map \
+        -H "Content-Type: application/json" \
+        -d '{"treasure_map_id": "f6ec73c93084ce91d5542a4ba6070071f5565112fe19b26ae9c960f9d658903a",
+             "bob_encrypting_key": "026d1f4ce5b2474e0dae499d6737a8d987ed3c9ab1a55e00f57ad2d8e81fe9e9ac"}'
+
+OR
+
+.. code:: bash
+
+    curl -X GET "<PORTER URI>/get_treasure_map?treasure_map_id=f6ec73c93084ce91d5542a4ba6070071f5565112fe19b26ae9c960f9d658903a&bob_encrypting_key=026d1f4ce5b2474e0dae499d6737a8d987ed3c9ab1a55e00f57ad2d8e81fe9e9ac"
+
+Example Response
+++++++++++++++++
+.. code::
+
+    Status: 200 OK
+
+
+.. code:: json
+
+    {
+       "result": {
+          "treasure_map": "Qld7S8sbKFCv2B8KxfJo4oxiTOjZ4VPyqTK5K1xK6DND6TbLg2hvlGaMV69aiiC5QfadB82w/5q1Sw+SNFHN2esWgAbs38QuUVUGCzDoWzQAAAGIAuhw12ZiPMNV8LaeWV8uUN+au2HGOjWilqtKsaP9fmnLAzFiTUAu9/VCxOLOQE88BPoWk1H7OxRLDEhnBVYyflpifKbOYItwLLTtWYVFRY90LtNSAzS8d3vNH4c3SHSZwYsCKY+5LvJ68GD0CqhydSxCcGckh0unttHrYGSOQsURUI4AAAEBsSMlukjA1WyYA+FouqkuRtk8bVHcYLqRUkK2n6dShEUGMuY1SzcAbBINvJYmQp+hhzK5m47AzCl463emXepYZQC/evytktG7yXxd3k8Ak+Qr7T4+G2VgJl4YrafTpIT6wowd+8u/SMSrrf/M41OhtLeBC4uDKjO3rYBQfVLTpEAgiX/9jxB80RtNMeCwgcieviAR5tlw2IlxVTEhxXbFeopcOZmfEuhVWqgBUfIakqsNCXkkubV0XS2l5G1vtTM8oNML0rP8PyKd4+0M5N6P/EQqFkHH93LCDD0IQBq9usm3MoJp0eT8N3m5gprI05drDh2xe/W6qnQfw3YXnjdvf2A="
+       },
+       "version": "6.0.0"
+    }
+
+
+POST /exec_work_order
+^^^^^^^^^^^^^^^^^^^^^
+Use a work order to execute a re-encrypt operation on the network network as part of Bob's ``retrieve`` workflow.
+
+Parameters
+++++++++++
++----------------------------------+---------------+----------------------------------------+
+| **Parameter**                    | **Type**      | **Description**                        |
++==================================+===============+========================================+
+| ``ursula``                       | String        | | Checksum address corresponding to    |
+|                                  |               | | the Ursula to execute the work order.|
++----------------------------------+---------------+----------------------------------------+
+| ``work_order_payload``           | String        | Work order payload encoded as base64.  |
```

Comment:
Where would one get it from? I don't think it's exposed to the user.

Reply:
Bob has to construct one. Is this not possible currently?

Reply:
It is possible, but it is sort of internal, and may change significantly after #2730. Currently one can obtain some work orders via `work_orders_for_capsules()` (which is public, but not documented), but there are no public methods that take them. What's the scenario here? Why would the user want work orders?

Reply:
The underlying scenario is what does Bob need to provide to Porter to get data re-encrypted by an Ursula, without disclosing anything sensitive...I assumed that was simply a WorkOrder payload that gets passed to the Ursula `/reencrypt` endpoint  See https://github.com/nucypher/nucypher/issues/2667. 

Do we need to make something public for work orders, or reconsider the strategy?

Reply:
Why not just make the whole `retrieve()` a Porter method? Why would Bob need a fine-grained control over work orders?

Reply:
Ah ok, `retrieve()` returns cleartexts. Then actually the `retrieve_cfrags()` method from #2730 would work quite well.

Reply:
Looks like `work_order_payload` would be the equivalent of the bytes representation of the `RemoteWorkOrder` object

Reply:
Does it need to? I don't see why we should expose work orders at all. `retrieve_cfrags()` seems to cover all the Bob's needs without exposing the actual cleartexts. The only thing that may be missing here is the ability to select Ursulas to contact, but I'm not sure if that's necessary.

Reply:
Unsure what you mean. Do you mean Porter should take the same parameters as `retrieve_cfrags()` and then perform the same functionality?

> The only thing that may be missing here is the ability to select Ursulas to contact, but I'm not sure if that's necessary.
Could you clarify this as well?

Reply:
Sorry, I just wanted to understand the requirements here to make sure I address them in 2730. For now this indeed seems to be the only way to retrieve via Porter without disclosing cleartexts to it. 

> The only thing that may be missing here is the ability to select Ursulas to contact, but I'm not sure if that's necessary.

So, currently, Bob can pick Ursulas he wants to contact for reencryption (since he provides work orders to Porter). If 2730 API (in its current form) is merged, Porter will only expose `retrieve_cfrags`, where Ursulas will be chosen by Porter, and Bob won't be able to affect it. I wonder if it is important to give Bob the ability to choose the Ursulas too.

Reply:
> I just wanted to understand the requirements here to make sure I address them in 2730

👍 . The philosophical requirement is that Porter shouldn't gain access to any sensitive material when performing a re-encryption on behalf of Bob.

> currently, Bob can pick Ursulas

A somewhat limited "pick" since the set of ursulas is from the treasure map, and each work order is associated with a single ursula - although I guess you mean to achieve the threshold - in that case yes. Based on the current code, I was effectively trying to avoid dealing with Capsules in Porter.

Seems like we can follow-up on what this endpoint may look like/change to in #2730 , if the retrieval process is going to be refactored/updated. If that gets cleared up, then modifying the endpoint is relatively easy by just updating the endpoint schema and refactoring the common retrieval code to be used both by Porter and in the pure Python context.